test/testdata/wsl.go

```diff
@@ -100,3 +100,75 @@ func f3() int {
 }
 
 func onelineShouldNotError() error { return nil }
+
+func multilineCase() {
+	// Multiline cases
+	switch {
+	case true,
+		false:
+		fmt.Println("ok")
+	case false ||
+		true:
+		fmt.Println("ok")
+	case true,
+		false:
+		fmt.Println("ok")
+	}
+}
+
+func sliceExpr() {
+	// Index- and slice expressions.
+	var aSlice = []int{1, 2, 3}
+
+	start := 2
+	if v := aSlice[start]; v == 1 {
+		fmt.Println("ok")
+	}
+
+	notOk := 1
+	if v := aSlice[start]; v == 1 { // ERROR "if statements should only be cuddled with assignments used in the if statement itself"
+		fmt.Println("notOk")
+		fmt.Println(notOk)
+	}
+
+	end := 2
+	if len(aSlice[start:end]) > 2 {
+		fmt.Println("ok")
+	}
+}
+
+func indexExpr() {
+	var aMap = map[string]struct{}{"key": {}}
+
+	key := "key"
+	if k, ok := aMap[key]; ok {
+		fmt.Println(k)
+	}
+
+	xxx := "xxx"
+	if _, ok := aMap[key]; ok { // ERROR "if statements should only be cuddled with assignments used in the if statement itself"
```

Comment:
but how to rewrite this code properly? it's the common coding pattern

Reply:
I see how that's a question for the linter itself and there's an issue [here](https://github.com/bombsimon/wsl/issues/249) to allow the variable to be cuddled if it's used anywhere in the block body. Before this PR the linter gave an error even if `xxx` was used as the index expression. :(

The way to rewrite to satisfy the linter:

```go
notUsedInIf := "separated from the statement"

if _, ok := aMap[key]; ok {
    fmt.Println("Not used here either")
    // Many lines later...
    fmt.Println(notUsedInIf)
}
```

And if the variable is used it may be cuddled.

```go
key := "used in if"
if _, ok := aMap[key]; ok {
    fmt.Println("ok")
}
```

Or if it's used after the condition.

```go
xxx := "not in if but first in block"
if _, ok := aMap[key]; ok {
    fmt.Println(xxx) // Ok
}
```

Reply:
I got a bit confused after writing my comment and had a reviset to [Effective Go](https://golang.org/doc/effective_go.html). I see that a lot of the code uses a pattern that WSL discourages in favour of readability. Although I don't see newlines being discussed in the documentation and the examples doesn't state it's the only (or preferred way) to handle empty lines.

The reason for me adding it to this project is that I got asked by two of the places I've worked who uses `golangci-lint` and uses this kind of style to improve readability. I got a bit anxious about it since it's mainly based on personal preferences and sent and e-mail to you but I didn't get any response so I didn't revert anything.

I know it's not much but since this got merged in v1.20.0 the linter have had 10 stars and a few issues added so there seems to be a bit of an interest. Although maybe not enough to be a part of this project?

Do you think this kind of linter/styling is out of scope for golangci-lint? If so, do you wan't me to revert it being added?

If you wan't more details and thoughts behind this linter please give the README from the repository a few minutes!

Reply:
got it, thank you!